### PR TITLE
feature : add read from levelinfo,complete add levelinfo to version,f…

### PR DESCRIPTION
**Pull request by Subsegment**
- all read method works correctly(temporary)
- flush im_memcache to tsm
- sync levelinfo version
- fix some bug #30 #29 #28

### CODE REVIEW

**Comment by Subsegment** [comment]
format了一下，force push

File: tskv/src/kvcore.rs [hunk 1]
```diff
@@ -142,35 +158,38 @@ impl TsKv {
         Ok(WritePointsRpcResponse { version: 1, points: vec![] })
     }
 
-    pub async fn read_point(&self,sid: SeriesID, time_range: &TimeRange, field_id: FieldID) {
+    pub async fn read_point(&self, sid: SeriesID, time_range: &TimeRange, field_id: FieldID) {
         let mut version_set = self.version_set.write().await;
-        let tsf= version_set.get_tsfamily(sid).unwrap();
-        //get data from memcache
+        let tsf = version_set.get_tsfamily(sid).unwrap();
```
Comment (zipper-meng) [inline tskv/src/kvcore.rs:163]: why not return empty result set when cannot get a tsfamily

File: tskv/src/tseries_family.rs [hunk 2]
```diff
@@ -209,18 +216,54 @@ impl TseriesFamily {
...
+                                         self.mut_cache.clone(),
+                                         self.immut_cache.clone(),
+                                         self.version.clone(),
+                                         self.opts.clone(),
+                                         self.super_version_id.load(Ordering::SeqCst));
+            self.super_version = Arc::new(vers);
+
+            if self.immut_cache.len() >= MAX_IMMEMCACHE_NUM {
+                println!("immut_cache num full,ready to send flush request");
+                let mut req_mem = vec![];
```
Comment (roseboy-liu) [inline tskv/src/tseries_family.rs:250]: refactor those code  ·put_mutcache·

File: tskv/src/kvcore.rs [hunk 3]
```diff
@@ -142,65 +158,85 @@ impl TsKv {
         Ok(WritePointsRpcResponse { version: 1, points: vec![] })
     }
 
-    pub async fn read_point(&self,sid: SeriesID, time_range: &TimeRange, field_id: FieldID) {
+    pub async fn read_point(&self, sid: SeriesID, time_range: &TimeRange, field_id: FieldID) {
         let mut version_set = self.version_set.write().await;
```
Comment (roseboy-liu) [inline tskv/src/kvcore.rs:162]: refactor those code ·read point·

**Review by roseboy-liu — APPROVED** [review]

**Pull request merged**